src/WebJobs.Script/Scale/FunctionsScaleManager.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Scale
+{
+    /// <summary>
+    /// Manages scale monitoring operations.
+    /// </summary>
+    public class FunctionsScaleManager
+    {
+        private readonly IScaleMonitorManager _monitorManager;
+        private readonly IScaleMetricsRepository _metricsRepository;
+        private readonly ILogger _logger;
+
+        // for mock testing only
+        public FunctionsScaleManager()
+        {
+        }
+
+        public FunctionsScaleManager(IScaleMonitorManager monitorManager, IScaleMetricsRepository metricsRepository, ILoggerFactory loggerFactory)
+        {
+            _monitorManager = monitorManager;
+            _metricsRepository = metricsRepository;
+            _logger = loggerFactory.CreateLogger(ScriptConstants.TraceSourceScale);
+        }
+
+        /// <summary>
+        /// Get the current scale status (vote) by querying all active monitors for their
+        /// scale status.
+        /// </summary>
+        /// <param name="context">The context to use for the scale decision.</param>
+        /// <returns>The scale vote.</returns>
+        public virtual async Task<ScaleStatusResult> GetScaleStatusAsync(ScaleStatusContext context)
+        {
+            var monitors = _monitorManager.GetMonitors();
+
+            List<ScaleVote> votes = new List<ScaleVote>();
+            if (monitors.Any())
+            {
+                // get the collection of current metrics for each monitor
+                var monitorMetrics = await _metricsRepository.ReadAsync(monitors);
+
+                _logger.LogInformation($"Computing scale status (WorkerCount={context.WorkerCount})");
+                _logger.LogInformation($"{monitorMetrics.Count} scale monitors to sample");
```

Comment:
How chatty is this? Should it be debug so we can see it in Kusto but it's not there by default for customers? Should we combine these into a single message?

Reply:
Changed to Debug logs